### PR TITLE
runtime-rs: fix default kernel location and add more default config paths

### PR DESCRIPTION
**Pull request by fidencio**
---

build: Use the correct kernel name

When calling `MAKE_KERNEL_NAME` we're considering the default kernel
name will be `vmlinux.container` or `vmlinuz.container`, which is not
the case as the runtime-rs, when used with dragonball, relies on the
`vmlinu[zx]-dragonball-experimental.container` kernel.

Other hypervisors will have to introduce a similar
`MAKE_KERNEL_NAME_${HYPERVISOR}` to adapt this to the kernel they want
to use, similarly to what's already done for the go runtime.

By doing this we also ensure that no changes in the configuration file
will be required to run runtime-rs, with dragonball, as part of our CI
or as part of kata-deploy.

Fixes: #6290

---

kata-types/config: Add more default config paths

This is not ideal, but this is what it is right now, as the
configurations end up being loaded from that list in case
`KATA_CONF_FILE` is not set.

As part of our CI, as part of kata-deploy, and as part of the payload we
provide to Confidential Containers, we do not set `KATA_CONF_FILE`, thus
expanding the list seems like the most reasonable approach to take, at
least till github.com/kata-containers/kata-containers#3961 gets solved.

While expanding the list, let's also make sure to add the
`-dragonball.toml` as first options as nothing guarantees that the
default configuration will be pointing to that one.

Fixes: #6291

---

### CODE REVIEW

File: src/libs/kata-types/src/config/default.rs
Comment (studychao) [inline src/libs/kata-types/src/config/default.rs:20]: Is creating a soft link from `/../../configuration.toml` to `../../configuration-dragonball.toml` during CI environment builtup process or user installation process a better way?

**Review by studychao — COMMENTED** [review]
Hi @fidencio , LGTM on the kernel name one.
And a little question on the default config path

File: src/runtime-rs/Makefile
Comment (Ji-Xinyou) [inline src/runtime-rs/Makefile:375]: Will this affect that the runtime-rs will support more hypervisor in the future? It seems that this assumes runtime-rs only works with dragonball (although it does now).

File: src/libs/kata-types/src/config/default.rs
Comment (fidencio) [reply src/libs/kata-types/src/config/default.rs:20]: So, it depends a lot.  CI does that, but kata-deploy and in the payload we provide to Confidential Containers we don't do that.

The reason for that is because the default runtime and the default VMM are still the go one and QEMU.  So, IOW, if we just link `configuration-dragonball.toml` to `configuration.toml` we break the current behaviour and the current assumption of the users.

File: src/runtime-rs/Makefile
Comment (studychao) [reply src/runtime-rs/Makefile:375]: This one has a `DB` postfix.
So I believe when CLH/QEMU/other hypervisors are supported, they could introduce xxxx_CLH/xxxx_QEMU

Comment (fidencio) [reply src/runtime-rs/Makefile:375]: What other VMMs will have to do is create a new `MAKE_KERNEL_NAME` for their own usage.  But that's something they'll have to do when they become supported (which is not the case now).

File: src/libs/kata-types/src/config/default.rs
Comment (studychao) [reply src/libs/kata-types/src/config/default.rs:20]: Will this change cause go runtime to use `configuration-dragonball.toml` since it's the first option? (If go runtime and rust runtime are both installed in some cases? I think that is a situation in kata-deploy if I remember correctly)

Comment (fidencio) [reply src/libs/kata-types/src/config/default.rs:20]: Yep, that's exactly the situation with kata-deploy, where we'll have both the go and the rust runtime "installed" at the same time.

The changes that I did were to avoid having any kind of conflict in that specific case (which is also the case for Confidential Containers).

If we just linked `configuration-dragonball.toml` toi `configuration.toml` the `kata` runtime class would pick the go runtime, to use with QEMU, but using the DB configuration, and that would simply explode. :-)

Is there something else that I could try to explain in a better way?

Comment (fidencio) [reply src/libs/kata-types/src/config/default.rs:20]: By the way, http://jenkins.katacontainers.io/job/kata-containers-dragonball-tests-ubuntu-20.04-PR/372/console now has passed the point of the issues that I had before this PR. \o/

File: src/runtime-rs/Makefile
Comment (Ji-Xinyou) [reply src/runtime-rs/Makefile:375]: > What other VMMs will have to do is create a new `MAKE_KERNEL_NAME` for their own usage. But that's something they'll have to do when they become supported (which is not the case now).

I see, then this may affect the work related to @jodh-intel and @pmores. But I think this is good.

File: src/libs/kata-types/src/config/default.rs
Comment (studychao) [reply src/libs/kata-types/src/config/default.rs:20]: ohhh I got it. I misunderstood that `DEFAULT_RUNTIME_CONFIGURATIONS ` will also be used in go-runtime. So I thought this will influence go runtiem cause they will also choose dragonball config as default. My mistake, sorry.

But if later CLH/QEMU introduce their support for runtime-rs, will this change here makes it hard for them to use their config? Maybe a FIXME or TODO comments leave here?

Comment (fidencio) [reply src/libs/kata-types/src/config/default.rs:20]: > But if later CLH/QEMU introduce their support for runtime-rs, will this change here makes it hard for them to use their config?

That's a valid point, but I sincerely hope we are able to get rid of this by having https://github.com/kata-containers/kata-containers/issues/3961 implemented.

Comment (fidencio) [reply src/libs/kata-types/src/config/default.rs:24]: I've updated the patch with a TODO comment.
Thanks for the review!

**Comment by studychao** [comment]
Thanks！LGTM：）

**Review by studychao — APPROVED** [review]

**Comment by studychao** [comment]
@fidencio seems like dragonball-ci is broken, maybe because how the kata test choose guest kernel, I'll also help to take a look at this.

**Comment by studychao** [comment]
it's kinda strange that this PR breaks k8s-seccomp.bats in both Dragonball CIs.. I will do a retest

File: src/libs/kata-types/src/config/default.rs
Comment (studychao) [reply src/libs/kata-types/src/config/default.rs:24]: Thanks!

**Comment by fidencio** [comment]
> it's kinda strange that this PR breaks k8s-seccomp.bats in both Dragonball CIs.. I will do a retest

I'd like to check whether it will also break with my current tests PR (I'm in the process of doing that, right now fixing power issues), if the tests pass with the tests PR better, and we just hold this one till we get the other one solved.

**Review by amshinde — APPROVED** [review]

File: src/libs/kata-types/src/config/default.rs
Comment (jodh-intel) [inline src/libs/kata-types/src/config/default.rs:17]: Nit: typo.

```suggestion
    /// * Drop the `-dragonball.toml` configs once
```

File: src/libs/kata-types/src/config/default.rs
Comment (jodh-intel) [inline src/libs/kata-types/src/config/default.rs:28]: Typo bug:

```suggestion
        "/opt/confidential-containers/share/defaults/kata-containers/configuration.toml",
```

File: src/runtime-rs/Makefile
Comment (jodh-intel) [inline src/runtime-rs/Makefile:206]: I know this is following the existing pattern, but this name is potentially a little confusing: I read "DB" as "database" initially :smile:

File: src/runtime-rs/Makefile
Comment (jodh-intel) [reply src/runtime-rs/Makefile:375]: Thanks for the ping @Ji-Xinyou.

I think this change is fine, but if we need to add further logic, I'd be in favour of making the `Makefile` call a utility script to return the name rather than having the logic in the `Makefile`. That would be clearer and easier to maintain.

**Review by jodh-intel — COMMENTED** [review]
Thanks @fidencio - a few comments...

**Review by GabyCT — APPROVED** [review]

**Comment by fidencio** [comment]
This PR depends-on https://github.com/kata-containers/tests/pull/5477, let's merge the tests one first and only after that we merge this one.

**Comment by fidencio** [comment]
I dropped the changed related to adding more default paths, as this is actually not needed.
The other changes are still needed, though, as we progress on merging CCv0 content to main.

**Pull request merged**
